LEAF_Request_Portal/js/workflow.js

```diff
@@ -5,734 +5,805 @@ var workflow;
+                fail: function (err) {
+                    console.log("Error: " + err);
+                },
+            });
+        }
+        if (step.stepModules != undefined) {
+            for (var x in step.stepModules) {
                 if (
-                  typeof workflowStepModule[e.data.step.stepID][i].trigger !==
-                  "undefined"
+                    modulesLoaded[
+                        step.stepModules[x].moduleName + "_" + step.stepID
+                    ] == undefined
                 ) {
-                  actionTriggered = true;
-                  workflowStepModule[e.data.step.stepID][i].trigger(
-                    function () {
-                      applyAction(data);
-                    }
-                  );
-                  break;
+                    modulesLoaded[
+                        step.stepModules[x].moduleName + "_" + step.stepID
+                    ] = 1;
+                    $(`#form_dep_extension${step.dependencyID}`)
+                        .html(`<div style="padding: 8px 24px 8px">
+                        <div style="background-color: white; border: 1px solid black; padding: 16px">
+                            <h2>Loading...</h2>
+                        </div>
+                        </div>`);
+                    $(`#form_dep_container${step.dependencyID} .button`).attr(
+                        "disabled",
+                        true
+                    );
+                    $.ajax({
+                        type: "GET",
+                        url:
+                            rootURL +
+                            "ajaxScript.php?a=workflowStepModules&s=" +
+                            step.stepModules[x].moduleName +
+                            "&stepID=" +
+                            step.stepID,
+                        dataType: "script",
+                        success: function () {
+                            workflowStepModule[step.stepID][
+                                step.stepModules[x].moduleName
+                            ].init(step, rootURL);
+                            $(
+                                `#form_dep_container${step.dependencyID} .button`
+                            ).attr("disabled", false);
+                        },
+                        fail: function (err) {
+                            console.log("Error: " + err);
+                        },
+                    });
                 }
-              }
-              if (!actionTriggered) {
-                applyAction(data);
-              }
-            } else {
-              applyAction(data);
             }
-          else {
-            applyAction(data);
-          }
-        };
+        }
 
-        if (actionPreconditionFunc !== undefined) {
-          actionPreconditionFunc(e.data, completeAction);
-        } else {
-          completeAction();
+        // legacy workflow modules based on dependencyIDs
+        for (var u in step.jsSrcList) {
```

Comment:
var to let

Reply:
I'll make these changes but I think they only showed up because of some formatting changes.